webknossos/webknossos/dataset/view.py

```diff
@@ -616,6 +571,25 @@ def _mag_view_bounding_box_at_creation(self) -> BoundingBox:
         assert self._mag_view_bbox_at_creation is not None
         return self._mag_view_bbox_at_creation
 
+    def _open(self) -> None:
+        if self._dataset is not None:
+            raise Exception("Cannot open view: the view is already opened")
+        else:
+            self._dataset = Dataset.open(
+                str(self._path)
+            )  # No need to pass the header to the wkw.Dataset
+
+    def _close(self) -> None:
+        if self._dataset is None:
+            raise Exception("Cannot close View: the view is not opened")
+        else:
+            self._dataset.close()
+            self._dataset = None
+
+    def _open_if_necessary(self) -> None:
+        if self._dataset is None:
+            self._open()
```

Comment:
I'd recommend to use a property for `_dataset` and rename it to `_wkw_dataset`, so it automatically gets opened when needed, and closing it in the destructor `__del__` (which can also be invoked explicitly using `del view` or just the dataset-cache using `del view._wkw_dataset`):

```suggestion
    @property
    def _wkw_dataset(self) -> None:
        if self._cached_wkw_dataset is None:
            _cached_wkw_dataset = Dataset.open(
                str(self._path)
            )  # No need to pass the header to the wkw.Dataset
        return self._cached_wkw_dataset
        
    @_wkw_dataset.deleter
    def _wkw_dataset(self) -> None:
        if self._cached_wkw_dataset is not None:
            self._cached_wkw_dataset.close()
            self._cached_wkw_dataset = None

    def __del__(self) -> None:
        del self._cached_wkw_dataset
```